### PR TITLE
refactor(server): version logic

### PR DESCRIPTION
**Pull request by jrasm91**
Use semver library for version compatibility checking and parsing.

### CODE REVIEW

**Review by zackpollard — APPROVED** [review]

File: server/src/services/database.service.spec.ts [hunk 1]
```diff
@@ -13,211 +12,170 @@ describe(DatabaseService.name, () => {
...
   });
 
   it('should work', () => {
     expect(sut).toBeDefined();
   });
 
-  describe.each([
-    [{ vectorExt: DatabaseExtension.VECTORS, extName: 'pgvecto.rs', minVersion: new Version(0, 1, 1) }],
-    [{ vectorExt: DatabaseExtension.VECTOR, extName: 'pgvector', minVersion: new Version(0, 5, 0) }],
-  ] as const)('init', ({ vectorExt, extName, minVersion }) => {
```
Comment (mertalev) [inline server/src/services/database.service.spec.ts:28]: The tests should still test everything with pgvector and pgvecto.rs. There's only one test for pgvector now from what I can see.

File: server/src/services/database.service.ts [hunk 2]
```diff
@@ -1,163 +1,134 @@
...
+    if (!current || !semver.satisfies(current, POSTGRES_VERSION_RANGE)) {
+      throw new Error(
+        `Invalid PostgreSQL version. Found ${version}, but needed ${POSTGRES_VERSION_RANGE}. Please use a supported version.`,
+      );
+    }
+
     await this.databaseRepository.withLock(DatabaseLock.Migrations, async () => {
-      await this.createVectorExtension();
-      await this.updateVectorExtension();
-      await this.assertVectorExtension();
```
Comment (mertalev) [inline server/src/services/database.service.ts:35]: Why did you inline all of this? It looks really messy this way.

**Review by mertalev — CHANGES_REQUESTED** [review]
This is a good (and necessary) change, but I think the reduced test coverage and messy database service `init` should be resolved.

File: server/src/services/database.service.ts (hunk 2 above)
Comment (jrasm91) [reply server/src/services/database.service.ts:35]: I removed all the logging and consolidated the shared variables, so it looks a lot better now. Can you take another look?

File: server/src/services/database.service.spec.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/services/database.service.spec.ts:28]: This is a unit test not a e2e test. Can you tell me what code path in the database service is not tested anymore?

File: server/src/services/database.service.ts [hunk 3]
```diff
@@ -1,38 +1,126 @@
...
+        Otherwise, please update the version of ${name} in the Postgres instance to a compatible version.`,
+  createFailed: ({ name, extension, otherName }: CreateFailedArgs) => `
+        Failed to activate ${name} extension.
+        Please ensure the Postgres instance has ${name} installed.
+
+        If the Postgres instance already has ${name} installed, Immich may not have the necessary permissions to activate it.
+        In this case, please run 'CREATE EXTENSION IF NOT EXISTS ${extension}' manually as a superuser.
+        See https://immich.app/docs/guides/database-queries for how to query the database.
+
+        Alternatively, if your Postgres instance has ${name}, you may use this instead by setting the environment variable 'DB_VECTOR_EXTENSION=${otherName}'.
```
Comment (mertalev) [inline server/src/services/database.service.ts:48]: `name` should be `otherName` here.

Comment (jrasm91) [reply server/src/services/database.service.ts:48]: Ah, good catch. Thank you.

**Review by danieldietzler — APPROVED** [review]
LGTM!

File: server/src/services/database.service.spec.ts (hunk 1 above)
Comment (mertalev) [reply server/src/services/database.service.spec.ts:28]: The checks for updating, enforcing version constraints and displaying the right extension in logs aren’t being tested. The tests would still pass if we only handled them for pgvecto.rs.

File: server/src/services/database.service.ts (hunk 2 above)
Comment (mertalev) [reply server/src/services/database.service.ts:35]: Much better now, thanks.

File: server/src/services/database.service.spec.ts (hunk 1 above)
Comment (jrasm91) [reply server/src/services/database.service.spec.ts:28]: I can add tests to verify the extension in the log message and some to verify the semver ranges work as expected

Comment (jrasm91) [reply server/src/services/database.service.spec.ts:28]: Alright, I've updated the tests to check semver and log messages for the extension name.

Comment (mertalev) [reply server/src/services/database.service.spec.ts:28]: Sweet, thanks!

**Review by mertalev — APPROVED** [review]

**Pull request merged**